src/borg/remote.py

```diff
@@ -856,6 +872,10 @@ def put(self, id, data, wait=True):
     def delete(self, id, wait=True):
         """actual remoting is done via self.call in the @api decorator"""
 
+    @api(since=parse_version('1.0.99'))
+    def barrier(self):
+        """actual remoting is done via self.call in the @api decorator"""
```

Comment:
There is no remoting here.
Just do a `self.call_many('barrier', [])` with a command documenting this kludge here explicitly? The empty calls would avoid going into the write path and popping off the call there without a rpc call. This would need a special mention in the guard at the start of call_many for empty calls, but that still seems cleaner than the current dance.